File: 백준/Silver/11723. 집합/집합.py
import sys

input=sys.stdin.readline
n=int(input())
s=set()
num=[i for i in range(1,21)]
for i in range(n):
    arr=list(map(str,input().split()))
    if len(arr)==1:
        if arr[0]=="all":
            s.update(num)
        elif arr[0]=="empty":
            s.clear()
        continue
        
    arr[1]=int(arr[1])
    if arr[0]=="add":
        s.add(arr[1])
    elif arr[0]=="remove":
        if arr[1] in s:s.discard(arr[1])
    elif arr[0]=="check":
        if arr[1] in s:print(1)
        else:print(0)
    elif arr[0]=="toggle":
        if arr[1] in s:s.remove(arr[1])
        else: s.add(arr[1])

File: 백준/Silver/11723. 집합/집합.java
import java.io.*;
import java.util.StringTokenizer;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
        StringTokenizer st = new StringTokenizer(br.readLine());
        int n = Integer.parseInt(st.nextToken());
        int[] S = new int[20];

        for (int i = 0; i < n; i++){
            st = new StringTokenizer(br.readLine());
            String operator = st.nextToken();
            if (operator.equals("all")) {
                for (int j = 0; j < 20; j++) S[j] = 1;
                continue;
            } else if (operator.equals("empty")){
                for (int j = 0; j < 20; j++) S[j] = 0;
                continue;
            }

            int x = Integer.parseInt(st.nextToken());

            if (operator.equals("add")){
                S[x-1] = 1;
            } else if (operator.equals("remove")){
                S[x-1] = 0;
            } else if (operator.equals("check")){
                if (S[x-1] == 1){
                    bw.write("1\n");
                } else{
                    bw.write("0\n");
                }
            } else if (operator.equals("toggle")){
                S[x-1] = Math.abs(S[x-1] - 1);
            }
        }
        bw.flush();
        bw.close();
    }
}
